src/Sylius/Behat/Context/Setup/CustomerContext.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Setup;
+
+use Behat\Behat\Context\Context;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\User\Repository\CustomerRepositoryInterface;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+final class CustomerContext implements Context
+{
+    const DEFAULT_CUSTOMER_FIRST_NAME = 'John';
+    const DEFAULT_CUSTOMER_LAST_NAME = 'Doe';
+
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @var CustomerRepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @var FactoryInterface
+     */
+    private $customerFactory;
+
+    /**
+     * @param SharedStorageInterface $sharedStorage
+     * @param CustomerRepositoryInterface $customerRepository
+     * @param FactoryInterface $customerFactory
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        CustomerRepositoryInterface $customerRepository,
+        FactoryInterface $customerFactory
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->customerRepository = $customerRepository;
+        $this->customerFactory = $customerFactory;
+    }
+
+    /**
+     * @Given the store has customer :firstName :lastName with email :email
+     */
+    public function theStoreHasCustomerWithEmail($firstName, $lastName, $email)
```

Comment:
Why not take ``:firstName`` and ``:lastName`` as one ``:name`` parameter and split it inside? I doesn't look bad in implementation, but it hurst [when reading](https://github.com/Sylius/Sylius/pull/4705/files#diff-7a458939fb02cd815c9f2108efba6699R12) ``"Frodo" "Baggins"`` instead of ``"Frodo Baggins"`` :|